dms/models/directory.py

```diff
@@ -194,6 +195,40 @@ class DmsDirectory(models.Model):
                 """,
     )
 
+    def check_access_rule(self, operation):
+        for record in self.filtered(
+            lambda x: (
+                (not x.is_root_directory and x.storage_id.save_type == "attachment")
+                or (x.is_root_directory and x.root_storage_id.save_type == "attachment")
+            )
+            and x.res_model
+        ):
+            model = record.res_model
+            try:
+                if record.res_id:
+                    self.env[model].browse(record.res_id).check_access_rule(operation)
+                else:
+                    self.env[model].check_access_rights(operation)
+            except AccessError:
+                raise AccessError(
+                    _("No tienes permisos de %s para acceder al registro %s,%s")
+                    % (operation, record.res_model, record.res_id)
+                )
+        key = "perm_%s" % operation
+        for record in self.filtered(lambda x: x.complete_group_ids):
+            if not record.complete_group_ids.filtered(lambda x: x[key]):
+                raise AccessError(
+                    _("No tienes permisos de %s para acceder al registro") % operation
+                )
+        return
+
+    def check_access_complete_group_ids(self, operation):
+        key = "perm_%s" % operation
+        for record in self:
+            if not record.complete_group_ids.filtered(lambda x: x[key]):
+                return False
+        return True
```

Comment:
This can be much simpler too:

```suggestion
        return bool(self.mapped('complete_group_ids').filtered(key))
```